### DISCUSSION TITLE
All containers of Kong Ingress controller is not coming up.

### DISCUSSION THREAD

**Discussion by jaikratsinghtariyal** [Kong/kubernetes-ingress-controller#1860] [Q&A] [answered]
Hello, I am trying to deploy kong G/W as ingress controller in Kuma with below command.

`kumactl install gateway kong | kubectl apply -f -`

Seeing below error

```
W0927 06:47:27.447520 1 client_config.go:614] Neither --kubeconfig nor --master was specified. Using the inClusterConfig. This might not work.
-------------------------------------------------------------------------------
Kong Ingress controller
Release:
Build:
Repository:
Go: go1.16.7
-------------------------------------------------------------------------------
time="2021-09-27T06:47:27Z" level=info msg="version of kubernetes api-server: 1.22" api-server-host="https://10.96.0.1:443" git_commit=8b5a19147530eaac9476b0ab82980b4088bbc1b2 git_tree_state=clean git_version=v1.22.2 major=1 minor=22 platform=linux/amd64
time="2021-09-27T06:47:27Z" level=fatal msg="failed to fetch publish-service: services \"kong-proxy\" is forbidden: User \"system:serviceaccount:kuma-gateway:kong-serviceaccount\" cannot get resource \"services\" in API group \"\" in the namespace \"kuma-gateway\"" service_name=kong-proxy service_namespace=kuma-gateway
```

![image](https://user-images.githubusercontent.com/40471286/134860710-c292fb9e-5a02-4eac-b931-402bcd134500.png)

Need some help here.

Thanks
Jaikrat Singh

**Comment by shahamit** [comment]
I was able to get the kong ingress controller (in dbless mode) running with k3d (which internally uses k3s as the kubernetes engine). Never tried kuma.

**Reply by jaikratsinghtariyal** [reply]
Thanks for the comment @shahamit .... but I am trying with Kuma and kong ingress controller is one of the component to achieve something end to end.

**Comment by shaneutt** [comment, marked as answer]
It appears your deployment is missing the RBAC permissions needed to make API requests:

```
time="2021-09-27T06:47:27Z" level=fatal msg="failed to fetch publish-service: services \"kong-proxy\" is forbidden: User \"system:serviceaccount:kuma-gateway:kong-serviceaccount\" cannot get resource \"services\" in API group \"\" in the namespace \"kuma-gateway\"" service_name=kong-proxy service_namespace=kuma-gateway
```

You might need to look at the bindings and permissions for the service account there and make adjustments.

I also wonder if you are or would start having any problems with your Kubernetes version of `v1.22` :thinking: 

Take note as per our version compatibility matrix that only 2.x+ (2.0 currently in beta) support `v1.22`:

https://docs.konghq.com/kubernetes-ingress-controller/2.0.x/references/version-compatibility/#kubernetes

Given the output it's not clear which version of the KIC you're using, what version are you using?

**Reply by jaikratsinghtariyal** [reply]
Yes ... looks like kubernetes 1.22 was the problem. It's working fine with 1.19.1.

I am using below command for KIC installation. Not sure which version it is installing.

`kumactl install gateway kong |kubectl apply -f -`

But thanks for the version compatibility link.

**Reply by shaneutt** [reply]
Sure no problem, glad that helped. Keep a look out for KIC 2.0 releasing in the near future with 1.22+ compatibility :+1:
